src/index.ts

```diff
@@ -97,7 +94,6 @@ class LastUpdatedStore {
 }
 
 class ReleaseChecker {
-  private readonly client: Client<true>;
   private readonly octokit = new Octokit({ auth: env.GITHUB_TOKEN });
```

Comment:
Since we are using `undici`, could you pass the `fetch` to Octokit [following this guide](https://github.com/octokit/octokit.js#fetch-missing)? Then we can remove the `engine.node` option in `package.json` and the app will probably work on Node 16 (Railway default) too.